### PR TITLE
Banner hover changes #160

### PR DESCRIPTION
**Pull request by Infixo**
Hover can only be tracked over buttons, so a button is added around CityName label.
This button hides the big one partially, so extra hook for a click is needed.

### CODE REVIEW

**Comment by alexeyOnGitHub** [comment]
I like seeing screenshots from proposed UI changes - they are helpful in demonstrating the idea :)

File: Assets/UI/WorldView/citybannermanager_CQUI.lua [hunk 1]
```diff
@@ -956,7 +962,6 @@ function Initialize_CQUI()
     LuaEvents.CQUI_SettingsInitialized.Add(CQUI_OnSettingsInitialized);
     LuaEvents.CQUI_SettingsUpdate.Add(CQUI_OnSettingsUpdate);
 
-    Events.CityRemovedFromMap.Add(CQUI_OnCityRemovedFromMap);    -- CQUI erase real housing from improvements data everywhere when a city removed from map
```
Comment (alimulhuda) [inline Assets/UI/WorldView/citybannermanager_CQUI.lua:959]: Deleting this line here does not seem related to the the change for this PR. What is the reason for this deletion?

Comment (alimulhuda) [reply Assets/UI/WorldView/citybannermanager_CQUI.lua:959]: Disregard this comment. I saw that it's related to some other bug fix that's already been approved and merged to trunk.

**Comment by alimulhuda** [comment]
If I understand correctly the idea is to reduce the area that gets used for the city mouseover display. So in the screenshot below, the old area for the mouseover is red and the new area is green. The mouseover in question is the one that shows the city worked tile and next culture expansion tile. Correct me if I'm wrong in my understanding here @Infixo 

image

File: Assets/UI/WorldView/citybannermanager_CQUI.lua [hunk 2]
```diff
@@ -115,8 +115,14 @@ function CQUI_Common_CityBanner_Initialize(self, playerID, cityID, districtID, b
         and bannerStyle == BANNERSTYLE_LOCAL_TEAM 
         and playerID == Game.GetLocalPlayer()) then
         -- Register the callbacks 
-        self.m_Instance.CityBannerButton:RegisterCallback( Mouse.eMouseEnter, CQUI_OnBannerMouseOver );
-        self.m_Instance.CityBannerButton:RegisterCallback( Mouse.eMouseExit,  CQUI_OnBannerMouseExit );
+        self.m_Instance.CityNameButton:RegisterCallback( Mouse.eMouseEnter, CQUI_OnBannerMouseOver );
+        self.m_Instance.CityNameButton:RegisterCallback( Mouse.eMouseExit,  CQUI_OnBannerMouseExit );
+		self.m_Instance.CityNameButton:SetVoid1(playerID);
+		self.m_Instance.CityNameButton:SetVoid2(cityID);
```
Comment (alimulhuda) [inline Assets/UI/WorldView/citybannermanager_CQUI.lua:121]:
```suggestion
```
Unnecessary function calls since we have them below as well. They only need to be set once. Tried it without these 2 lines and it's working the same way.

File: Assets/UI/WorldView/citybannermanager_CQUI.lua [hunk 3]
```diff
@@ -115,8 +115,14 @@ function CQUI_Common_CityBanner_Initialize(self, playerID, cityID, districtID, b
...
+        -- Re-register normal click as it gets hidden by a new button
+        self.m_Instance.CityNameButton:RegisterCallback( Mouse.eLClick, OnCityBannerClick );
+		self.m_Instance.CityNameButton:SetVoid1(playerID);
+		self.m_Instance.CityNameButton:SetVoid2(cityID);
```
Comment (alimulhuda) [inline Assets/UI/WorldView/citybannermanager_CQUI.lua:125]:
```suggestion
        self.m_Instance.CityNameButton:SetVoid1(playerID);
        self.m_Instance.CityNameButton:SetVoid2(cityID);
```
Change from tabs to regular spaces so the indentation lines up nicely.

**Comment by Infixo** [comment]
Sorry for not replying earlier, different time zone.
It is hard to show on the screen the effect because there is no difference  in what is shown only when it is shown. The problem is that citizen icons in the lens are overlapping with the tooltips from the banner. This change makes it so the lens is only activated when there is no tooltip, so no more interference.
Thank you for reviewing it.
